pkg/executables/cmk.go

```diff
@@ -0,0 +1,363 @@
+package executables
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	b64 "encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"gopkg.in/ini.v1"
+
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+	"github.com/aws/eks-anywhere/pkg/templater"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+const (
+	cmkConfigFileName             = "cmk_tmp.ini"
+	cloudStackb64EncodedSecretKey = "CLOUDSTACK_B64ENCODED_SECRET"
+	cloudmonkeyInsecureKey        = "CLOUDMONKEY_INSECURE"
+)
+
+var (
+	//go:embed config/cmk.ini
+	cmkConfigTemplate string
+	requiredEnvsCmk   = []string{cloudStackb64EncodedSecretKey, cloudmonkeyInsecureKey}
+)
+
+const (
+	maxRetriesCmk    = 5
+	backOffPeriodCmk = 5 * time.Second
+)
+
+type Cmk struct {
+	writer     filewriter.FileWriter
+	executable Executable
+	retrier    *retrier.Retrier
+	execConfig *cmkExecConfig
+}
+
+func (c *Cmk) SearchTemplate(ctx context.Context, domain string, zone string, account string, template string) (string, error) {
+	templates, err := c.ListTemplates(ctx, template)
+	if err != nil {
+		return "", fmt.Errorf("search template %s error: %v", template, err)
+	} else if len(templates) > 1 {
+		return "", fmt.Errorf("duplicate templates %s found", template)
+	} else if len(templates) == 0 {
+		return "", fmt.Errorf("template %s not found", template)
```

Comment:
nit: no need to change but you could return `"", nil` here and let the caller decide if not finding the the template is an error or not

If you prefer to leave it as is, maybe you could rename the method to `FindTemplate`. I think that's a bit more explicit about expecting the template to exist.

Reply:
Agreed, the current contract/naming are unclear and confusing. I'm leaning towards renaming these methods to FindXXXX and re-defining the contract accordingly